src/lib/InstantSearch.js

```diff
@@ -74,6 +83,12 @@ Usage: instantsearch({
     }
 
     this.urlSync = urlSync === true ? {} : urlSync;
+    if (routing === true) this.routing = ROUTING_DEFAULT_OPTIONS;
```

Comment:
It looks like we can use both system at the same time, could be nice to throw an error when both options are provided.